e3sm_diags/driver/utils/io.py

```diff
@@ -68,13 +70,17 @@ def _save_data_metrics_and_plots(
         "long_name", "No long_name attr in test data"
     )
 
-    plot_func(
+    # Get the function arguments and pass to the set's plotting function.
+    args = [
         parameter,
         ds_test[var_key],
         ds_ref[var_key] if ds_ref is not None else None,
         ds_diff[var_key] if ds_diff is not None else None,
-        metrics_dict,
-    )
+    ]
+    if metrics_dict is not None:
+        args = args + [metrics_dict]
+
+    plot_func(*args)
```

Comment:
Updated `_save_data_metrics_and_plots()` to make `metrics_dict` optional for the `plot_func` (the set's plot function).

I pulled this from my PR #748.